src/string/utf16.cr

```diff
@@ -53,7 +53,15 @@ class String
   # slice = Slice[104_u16, 105_u16, 32_u16, 55296_u16, 56485_u16]
   # String.from_utf16(slice) # => "hi 𐂥"
   # ```
-  def self.from_utf16(slice : Slice(UInt16)) : String
+  #
+  # If *slice* is a pointer, the string is interpreted as null-terminated.
```

Comment:
The input pointer could be interpreted as a string, but maybe it's better to say that it ends when a value of zero is found?